JM/CF/Scripts/1_Core/CommunityFramework/CF_Uint.c

```diff
@@ -0,0 +1,138 @@
+/*
+	Uint Type Definition
+	--	This is a CF_Uint datatype definition for EnScript
+
+	Author: Lystic, Jacob_Mango
+
+	When working with uints, Right Shift operations need to be handled by the ShiftRight method.
+*/
+
+typedef int CF_Uint;
+class CF_Uint : int
+{
+	CF_Uint Add(CF_Uint b)
+	{
+		CF_Uint a = value;
+		CF_Uint carry = a & b;
+		CF_Uint result = a.XOR(b);
+		while (carry != 0)
```

Comment:
You can just do `while(carry)` because any value other than 0 is implicitly true. Saves the comparison on each iteration.